pkg/didcomm/common/service/event.go

```diff
@@ -61,14 +61,17 @@ type DIDCommAction struct {
 	// to stop the processing.
 	Stop func(err error)
 
-	// Properties contains value based on specific protocol. The consumers need to cast this to specific interface
-	// based on the protocol event.
+	// Properties contains value based on specific protocol. The consumers need to call the protocol client
+	// functions to get the data.
 	//
-	// Cast to following interfaces based on protocol
-	//   - DID Exchange :  service.DIDExchangeEvent
-	Properties interface{}
+	// Clients function to retrieve data based on protocol.
+	//   - DID Exchange :  didexchange.EventProperties
+	Properties EventProperties
 }
 
+// EventProperties type for event related data.
+type EventProperties interface{}
```

Comment:
This does nothing, we still have the interface. Just more confusing and dependency in the client pkg e.g `EventProperties(e service.EventProperties)`

Reply:
This is consumed by user and they invoke client.EventProperties util to fetch the actual details. This way the framework would have a way to change the actual type without affecting the users.